vector-im/element-android
# [Poll] Error handling for push rules synchronization

## Type of change


## Content

Display an error in the settings screen when push rules are out of sync (after opening the screen or modifying the rules)

Added a live refresh of the notification setting screens when the account data is updated (eg. from another client)

## Motivation and context

Error handling of the new push rules introduced in #8114

## Screenshots / GIFs

<img width="473" alt="image" src="https://user-images.githubusercontent.com/11990514/219609508-7267cef7-2e39-42be-961b-540767346821.png">

## Tests

On a synapse supporting the new push rules
- update the default rules in the notification settings (Direct messages and/or group messages fields)
- break the sync of the rules by editing the push rules entry in the account data (using `/devtools` from a web instance)
- verify that an error is displayed upon the broken rule

Note that the rules should be automatically realigned by #8130 so the error may be not visible

## Tested devices

- OS version(s):

## Checklist
